daemons/ipa-kdb/tests/ipa_kdb_tests.c

```diff
@@ -478,6 +494,44 @@ void test_dom_sid_string(void **state)
 }
 
 
+void test_check_trusted_realms(void **state)
+{
+    struct test_ctx *test_ctx;
+
+    test_ctx = (struct test_ctx *) *state;
+
+    for(size_t i = 0; i < NUM_SUFFIXES; i++) {
+        char *test_realm = NULL;
+        char *trusted_realm = NULL;
+        asprintf(&test_realm, TEST_REALM_TEMPLATE, i);
+	krb5_error_code kerr = 0;
+
+        if (test_realm) {
+            kerr = ipadb_is_princ_from_trusted_realm(
+                       test_ctx->krb5_ctx,
+                       test_realm,
+                       strlen(test_realm),
+                       &trusted_realm);
+            assert_int_equal(kerr, 0);
+            free(test_realm);
+            free(trusted_realm);
+        }
+
+        asprintf(&test_realm, EXTERNAL_REALM, i);
```

Comment:
The variable i is unused because the are no replacements in the EXTERNAL_REALM template.

Reply:
I changed this part to simply use external realm reference directly. Also moved it out of the loop as there is no need to build the variable anymore.